src/reverse.ts

```diff
@@ -0,0 +1,33 @@
+import { purry } from "./purry";
+
+/**
+ * Reverses array.
+ * @param array the array
+ * @signature
+ *    R.reverse(arr);
+ * @example
+ *    R.reverse([1, 2, 3]) // [3, 2, 1]
+ * @data_first
+ * @category Array
+ */
+export function reverse<T>(array: readonly T[]): Array<T>;
+
+/**
+ * Reverses array.
+ * @param array the array
+ * @signature
+ *    R.reverse()(array);
+ * @example
+ *    R.reverse()([1, 2, 3]) // [3, 2, 1]
+ * @data_last
+ * @category Array
+ */
+export function reverse<T>(): (array: readonly T[]) => Array<T>;
+
+export function reverse() {
+  return purry(_reverse, arguments);
+}
+
+function _reverse(array: any[]) {
+  return array.slice(0).reverse();
```

Comment:
Using spread operator for copying the array provides better readability and doesn't require using `magic number / zero`.

Reply:
The thing is they're not equivalent. There's at least one edge-case I know about:

```
const arr = []
arr.length = 2;
[...arr] // => [undefined, undefined]
```

slice() will return empty array 

I would normally use spread operator for the project I have a control over, but not for the lib like this.

Reply:
Good point. Still one improvement is possible, `slice` could be called without args since 0 is the default value.

Reply:
@leilapearson By changing the `length` you didn't put the `undefined` values into the array but `empty slots`, so why putting `undefined` values in the cloned copy doing `[...arr]` when the original array has `empty slots` in those indexes. In general, another big difference is that empty slots are non-iterable, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/length.

Reply:
Correcting myself after looking at it a bit more closely. When @sobstel said it would return an empty array I thought it was losing array sparseness in the process, but it appears not... and furthermore empty items are not the same thing as undefined items. 

```ts
console.log(arr); // [ 1 ]
console.log([...arr]); // [ 1 ]

arr.length = 3;

console.log(arr); // [ 1, <2 empty items> ]
console.log([...arr]); // [ 1, undefined, undefined ]
console.log([1, , ,]); // [1, <2 empty items> ]

console.log(arr.slice()); // [1, <2 empty items> ]
```
So you're right  @sobstel (naturally :-) that `arr.slice()` is the way to go.

Reply:
Thanks, @vladimiry. Didn't see your comment before typing my second comment. Great to learn something new :-)